pennylane/compiler/python_compiler/transforms/quantum/diagonalize_measurements.py

```diff
@@ -118,7 +128,9 @@ class DiagonalizeFinalMeasurementsPass(passes.ModulePass):
     def apply(self, _ctx: context.Context, module: builtin.ModuleOp) -> None:
         """Apply the diagonalize final measurements pass."""
         pattern_rewriter.PatternRewriteWalker(
-            pattern_rewriter.GreedyRewritePatternApplier([DiagonalizeFinalMeasurementsPattern()])
+            pattern_rewriter.GreedyRewritePatternApplier(
+                [DiagonalizeFinalMeasurementsPattern()],
+            ),
```

Comment:
We found out recently that the `GreedyRewritePatternApplier` is only useful if your pass applies multiple rewrite patterns.

```suggestion
            DiagonalizeFinalMeasurementsPattern()
```